src/InitializerTrait.php

```diff
@@ -27,12 +27,39 @@ trait InitializerTrait
     /**
      * Initialize object. Always call parent::init(). Do not call directly.
      */
-    public function init(): void
+    protected function init(): void
     {
         if ($this->_initialized) {
             throw (new Exception('Attempting to initialize twice'))
                 ->addMoreInfo('this', $this);
         }
         $this->_initialized = true;
     }
+
+    /**
+     * Do not call directly.
+     */
+    public function invokeInit(): void
+    {
+        // make sure init() method is never declared as public
+        $reflMethod = new \ReflectionMethod($this, 'init');
```

Comment:
checked for every instance, but fast: +0.5% of total testing time for atk4/ui where used heavily